web/src/user_status_ui.ts

```diff
@@ -1,19 +1,83 @@
 import $ from "jquery";
+import assert from "minimalistic-assert";
+import type * as tippy from "tippy.js";
 
 import render_set_status_overlay from "../templates/set_status_overlay.hbs";
 import render_status_emoji_selector from "../templates/status_emoji_selector.hbs";
 
 import * as dialog_widget from "./dialog_widget";
+import * as dropdown_widget from "./dropdown_widget";
 import * as emoji from "./emoji";
 import type {EmojiRenderingDetails} from "./emoji";
+import * as flatpickr from "./flatpickr";
 import {$t, $t_html} from "./i18n";
 import * as keydown_util from "./keydown_util";
 import * as people from "./people";
+import * as timerender from "./timerender";
 import * as user_status from "./user_status";
 import type {UserStatusEmojiInfo} from "./user_status";
 
 let selected_emoji_info: Partial<UserStatusEmojiInfo> = {};
 let default_status_messages_and_emoji_info: {status_text: string; emoji: EmojiRenderingDetails}[];
+let user_status_widget: dropdown_widget.DropdownWidget;
+let is_custom_time_selected: boolean;
+let status_end_time: number | undefined;
+let scheduled_timestamp: string;
+let custom_time_selected: number;
+
+type ClearUserStatusOption = {
+    name: string;
+    unique_id: string;
+};
+
+type UserStatusData = {
+    status_text: string;
+    emoji_name: string;
+    emoji_code: string;
+    reaction_type: string;
+};
+
+const clear_user_status_option_vals: ClearUserStatusOption[] = [
+    {name: "Never", unique_id: "1"},
+    {name: "In one hour", unique_id: "2"},
+    {name: "Today at 5:00 PM", unique_id: "3"},
+    {name: "Tomorrow at 9:00 AM", unique_id: "4"},
+    {name: "Custom", unique_id: "5"},
+];
```

Comment:
These strings should all be tagged for translation. I would also recommend making the unique keys nicer.

See `scheduled_messages.ts` for an example nicer system for populating times for this sort of menu.

Reply:
I tagged the strings for translations, and regarding the `unique_id`, I made them into numbers, and updated the pr.

The `scheduled_messages` doesn't use a dropdown, so the structure would be totally different from this... However, I'll try to clean this up a bit.